Add pkgin installer.

weakish opened the pull request:
`pkgin` is a binary package manager for pkgsrc on BSD, macOS,
Linux, SunOS, Minix and SmartOS/Illumos.

---
 lib/sprinkle/installers/pkgin.rb | 39 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)
 create mode 100644 lib/sprinkle/installers/pkgin.rb

yyyc514 commented:
Looks pretty good.

--

Can pkgin not install multiple packages at the same time by listing them out?

Should use sudo_cmd, not explicitly invoke sudo.

weakish commented:
> Can pkgin not install multiple packages at the same time by listing them out?

Yes.
Refactored to use `auto_api`.

> Should use sudo_cmd

Fixed.

yyyc514 commented on lib/sprinkle/installers/pkgin.rb:
```diff
@@ -10,29 +10,30 @@ module Installers
-      end
       verify_api do
         def has_pkgin(package)
           @commands << "pkgin list | egrep '^#{@package_name}-'"
```
Didn't we decide this should be `grep -e`?

weakish replied:
Sorry. New commits fixed this typo.
